### PR TITLE
[Core: User] Fixed Type Check for hasCenter Method - HACKTOBERFEST 2018 🎃

### PR DESCRIPTION
**Pull request by HenriRabalais**
### Previously on User...
The `hasCenter` method enforces strict type checking, however in the function doc it explicitly says that it expects an `int`. This `int` is then forced to be compared with a `string` provided by the `getCenterIDs` method. Therefore, if an `int` is passed to the function, the result will necessarily be false.

### This Week on FINAL HACKTOBERFEST PR:
I cast the value being stored in the CenterIDs array to ints. This allows a strict comparison between the values passed to `hasCenterID` and the `getCenterIDs` array.

### CODE REVIEW

File: php/libraries/User.class.inc [hunk 1]
```diff
@@ -76,7 +76,7 @@ class User extends UserPermissions
         );
         $user_cid            = array();
         foreach ($user_centerID_query as $key=>$val) {
-            $user_cid[$key] =$val['CenterID'];
+            $user_cid[$key] = intval($val['CenterID']) //Cast values to int;
```
Comment (johnsaigle) [inline php/libraries/User.class.inc]: Pedantic note but technically this isn't the same as casting so I changed the comment.

More importantly you commented out you semi-colon here so this isn't gonna work!

```suggestion
            // Treat string representation of ID as an int
            $user_cid[$key] = intval($val['CenterID']);
```

Comment (johnsaigle) [reply php/libraries/User.class.inc]: Sure. I updated my suggestion so you can commit it more easily.

Comment (HenriRabalais) [reply php/libraries/User.class.inc]: Committed

**Comment by kongtiaowang** [comment]
LGTM

**Pull request merged**